### PR TITLE
[v7] Refactor Drone Pipelines to use AWS role assumption

### PR DESCRIPTION
**Pull request by wadells**
Backports https://github.com/gravitational/teleport/pull/17201
Backports https://github.com/gravitational/teleport/pull/17274
Backports https://github.com/gravitational/teleport/pull/17314

Contributes to https://github.com/gravitational/SecOps/issues/213

There were a large range of merge conflicts during this port, including:

v9+ conflicts:
* Windows native builds aren't on v9 and prior
* Teleport Connect isn't in v9 and prior, including some refactoring of the mac pipelines done to enable mac connect builds.

v7 specific conflicts:
* Logan's ECR work isn't in v7, meaning there are lots of places that don't need AWS, as they only push to Quay
* Fred's new rpm/deb work isn't in v7, so all commits (and fixes) related to this could be dropped)
* For the two buildbox changes, are published to quay, on v7, but I took these anyhow as they included tidying changes across all pipelines.  This will help future porting.

## Testing Done
Definitely needed given the wide array of merge conflicts.
* tag: https://drone.platform.teleport.sh/gravitational/teleport/16477
* promote: https://drone.platform.teleport.sh/gravitational/teleport/16484

### CODE REVIEW

**Review by r0mant — APPROVED** [review]
lgtm assuming you've tested it.

**Comment by wadells** [comment]
> lgtm assuming you've tested it.

Still working through that.  I won't merge until I have a successful build and promote.

**Review by reedloden — APPROVED** [review]

**Comment by wadells** [comment]
Tag & promote are clean -- merging.

**Pull request merged**
